src/sv-pipeline/scripts/validate_ped.py

```diff
@@ -0,0 +1,137 @@
+    for sub in ILLEGAL_ID_SUBSTRINGS:
+        if sub in identifier:
+            raise ValueError(f"Invalid {id_type} ID in {source_file}: {identifier}. " +
+                             f"IDs cannot contain the following substrings: {', '.join(ILLEGAL_ID_SUBSTRINGS)}.")
+
+    # check for duplicate IDs
+    if id_set is not None:
+        if identifier in id_set:
+            raise ValueError(f"Duplicate {id_type} ID in {source_file}: {identifier}")
+
+
+def get_samples(samples_file):
+    # Note that this sample ID validation is incomplete:
+    # * It does not check if sample IDs are substrings of other sample IDs
+    # * It only checks the provided sample list for duplicates, which is not the full cohort in GatherBatchEvidence
+    samples = set()
+    with open(samples_file, 'r') as samp:
+        for line in samp:
+            sample = line.strip()
+            validate_id(sample, samples, "sample", "sample list")
+            samples.add(sample)
+
+    return samples
+
+
+def validate_ped(ped_file, samples):
+    seen_sex_1 = False
+    seen_sex_2 = False
+    samples_found = set()
+    with open(ped_file, 'r') as ped:
+        for line in ped:
+            # specification allows commented lines, which should be removed by SubsetPedFile task
+            if line.startswith("#"):
+                continue
+
+            # we require tab-delimited PED files although the specification does not
+            fields = line.strip().split("\t")
+            if len(fields) < 6:
+                raise ValueError("Invalid PED file. PED file must be tab-delimited and have 6 columns: " +
+                                 "family ID, sample ID, paternal ID, maternal ID, sex, and phenotype.")
+
+            # validate IDs
+            # don't check for duplicates here:
+            # family and parent IDs may appear multiple times, and sample IDs checked elsewhere
+            for identifier, id_type in zip(fields[:FIELD_NUMBER_SEX], ["family", "sample", "parent", "parent"]):
+                if id_type == "parent" and identifier == "0":
+                    continue  # allow maternal and paternal ID to be 0 (otherwise all-numeric IDs not allowed)
+                validate_id(identifier, None, id_type, "PED file")
+
+            # check for at least one appearance each of 1 and 2 in sex column
+            # this check is to ensure M/F are coded as 1 and 2 (not 0 and 1)
+            # and both M and F are present (for sex-specific steps)
+            sample_id = fields[FIELD_NUMBER_ID]
+            sex = fields[FIELD_NUMBER_SEX]
+            if sex == "1":
+                seen_sex_1 = True
+            elif sex == "2":
+                seen_sex_2 = True
+
+            # require integer sex
+            elif not sex.isdigit():
+                raise ValueError(f"Sample {sample_id} has an invalid value for sex: {sex}. " +
+                                 "PED file must use integer values for sex: 1=Male, 2=Female, 0=Unknown/Other.")
```

Comment:
Should we be more strict and require that it be 0, 1, or 2?

Reply:
Either way is fine with me. I did it this way because it will only fail in CleanVcfPart1 if it's not an integer, and I wasn't sure if some groups might want to encode different categories of "other" in the sex column. But it does seem simpler to just enforce 0,1,2